src/packagedcode/licensing.py

```diff
@@ -140,6 +140,9 @@ def add_referenced_license_matches_for_package(resource, codebase):
             ):
                 continue
 
+            if not (referenced_resource and referenced_resource.license_detections):
```

Comment:
Actually, referenced_resource is referenced below outside of this loop and may not exists at all.